src/AnalysisPrograms/Recognizers/GenericRecognizer.cs

```diff
@@ -405,6 +447,51 @@ public class GenericRecognizerConfig : RecognizerConfig, INamedProfiles<object>
         {
             /// <inheritdoc />
             public Dictionary<string, object> Profiles { get; set; }
+
+            // ########### THE FOLLOWING PROPERTIES ARE FOR POST-PROCESSING OF EVeNTS.
+
+            /// <summary>
+            /// Gets or sets a value indicating Whether or not to combine overlapping events.
+            /// </summary>
+            public bool CombineOverlappingEvents { get; set; }
+
+            /// <summary>
+            /// Gets or sets a value indicating Whether or not to combine events that constitute a sequence of the same strophe.
+            /// </summary>
+            public bool CombinePossibleSyllableSequence { get; set; }
+
+            /// <summary>
+            /// Gets or sets a value indicating the maximum allowable start time gap (seconds) between events within the same strophe.
+            /// This value is used only where CombinePossibleSyllableSequence = true.
+            /// </summary>
+            public double SyllableStartDifference { get; set; }
+
+            /// <summary>
+            /// Gets or sets a value indicating the maximum allowable difference (in Hertz) between the frequency bands of two events. I.e. events should be in similar frequency band.
+            /// NOTE: SIMILAR frequency band means the differences between two top Hertz values and the two low Hertz values are less than hertzDifference.
+            /// This value is used only where CombinePossibleSyllableSequence = true.
+            /// </summary>
+            public double SyllableHertzGap { get; set; }
+
+            // #### The next three properties determine filtering of events based on acoustic conctent of upper and lower buffer zones.
+
+            /// <summary>
+            /// Gets or sets a value indicating Whether or not to filter events based on acoustic conctent of upper buffer zone.
+            /// If value = 0, the upper neighbourhood is ignored.
+            /// </summary>
+            public int NeighbourhoodUpperHertzBuffer { get; set; }
+
+            /// <summary>
+            /// Gets or sets a value indicating Whether or not to filter events based on the acoustic content of their lower buffer zone.
+            /// If value = 0, the lower neighbourhood is ignored.
+            /// </summary>
+            public int NeighbourhoodLowerHertzBuffer { get; set; }
+
+            /// <summary>
+            /// Gets or sets a value indicating the decibel threshold for acoustic activity in the upper and lower buffer zones.
+            /// This value is used only if NeighbourhoodLowerHertzBuffer > 0 OR NeighbourhoodUpperHertzBuffer > 0.
+            /// </summary>
+            public double NeighbourhoodDbThreshold { get; set; }
```

Comment:
this should re-designed, but that can be done later

Reply:
OK.